\documentclass{uofsthesis-cs}

% Documentation for the uofsthesis-cs class is given in uofsthesis-cs.dvi
% 
% It is recommended that you read the CGSR thesis preparation
% guidelines before proceeding.
% They can be found at http://www.usask.ca/cgsr/thesis/index.htm

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% FRONTMATTER - In this section, specify information to be used to
% typeset the thesis frontmatter.
\usepackage{graphicx} 
\usepackage{cite}   
\usepackage{url}  
\usepackage{todonotes}
\usepackage{textcomp}
\usepackage{hyperref}

\pdfminorversion=5 
\pdfcompresslevel=9
\pdfobjcompresslevel=2
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

% THESIS TITLE
% Specify the title. Set the capitalization how you want it.
\title{SynVisio: A Multiscale Tool to Explore Genomic Conservation}

% AUTHOR'S NAME
% Your name goes here.
\author{Venkat Kiran Bandi}

% DEGREE SOUGHT.  
% Use \MSc or \PhD here
\degree{\MSc}         

% THESIS DEFENCE DATE
% Should be month/year, e.g. July 2004
\defencedate{May/2020}


% NAME OF ACADEMIC UNIT
%
% The following two commands allow you to specify the academic unit you belong to.
% This will appear on the title page as
% ``<academic unit> of <department>''.
% So if you are in the division of biomedical engineering you would need to do:
% \department{Biomedical Engineering}
% \academicunit{Division}
%
% The default is ``Department of Computer Science'' if these commands
% are not given.
%
% If you are in a discipline other than Computer Science, uncomment the following line and
% specify your discipline/department.  Default is 'Computer Science'.
% \department{If not Computer Science, put the name of your department here}

% If you are not in a department, but say, a division, uncomment the following line.
% \academicunit{Put the type of academic unit you belong to here, e.g. Division, College}


% PERMISSION TO USE ADDRESS
%
% If you are not in Comptuer Science you will want to change the
% address on the Permission to Use page.  This is done using the
% \ptuaddress{}.  Example:
%
% \ptuaddress{Head of the Department of Computer Science\\
% 176 Thorvaldson Building\\
% 110 Science Place\\
% University of Saskatchewan\\
% Saskatoon, Saskatchewan\\
% Canada\\
% S7N 5C9
% }

% ABSTRACT
\abstract{
Comparative analysis of genomes is an important area in biological research that can shed light on an organism's internal functions and evolutionary history. It involves comparing two or more genomes to identify similar regions that can indicate shared ancestry and in turn conservation of genetic information. Due to rapid advancements in sequencing systems, high-resolution genome data is readily available for a wide range of species, and comparative analysis of this data can offer crucial  evolutionary insights that can be applied in plant breeding and medical research. Visualizing the location, size, and orientation of conserved regions can assist research in comparative analysis as it is a tedious process that requires extensive manual interpretation and human judgement. However, visualization tools for the analysis of conserved regions have not kept pace with the increasing availability of information and are not designed to support the diverse use cases of biological researchers. To address this we gathered feedback from experts in the field, and developed improvements for these tools through novel interaction techniques and visual representations. We then developed SynVisio, a web-based tool for exploring conserved regions at multiple resolutions (genome, chromosome, or gene), with several visual representations and interactive features, to meet the diverse needs of genome researchers. SynVisio supports multi-resolution analysis and interactive filtering as researchers move deeper into the genome. It also supports revisitation to specific interface configurations, and enables loosely-coupled collaboration over the genomic data.  An evaluation of the system with five researchers from three expert groups coupled with a longitudinal study of web traffic to the system provides evidence about the success of our system's novel features for interactive exploration of genomic conservation.
}

% THESIS ACKNOWLEDGEMENTS -- This can be free-form.

\acknowledgements{

I would like to express my heartfelt gratitude to my supervisor Carl Gutwin, for constantly offering me support both academically and personally, and guiding me towards a field of research that was fantastic to explore and learn.

I am particularly grateful to Gwen Lancaster for her support when I joined the program and am also grateful to my lab mates and all the staff members of the Department of Computer Science who have helped me along the way. Also, I would like to thank the members of my thesis committee, Ian McQuillan and Debajyoti Mondal, whose comments and suggestions have greatly improved this manuscript. 

Finally, I would like to thank Canada for offering me a chance at a fresh start and I acknowledge the support of my family and friends for helping me along this journey in starting a new life.

}

% THESIS DEDICATION -- Also free-form.  If you don't want a dedication, comment out the following
% line.
\dedication{

\begin{center}
\textit{This thesis is dedicated to my Mom and Dad for their unconditional love and support.}
\end{center}

}

% LIST OF ABBREVIATIONS - Sample  
% If you don't want a list of abbreviations, comment the following 4 lines.
\loa{
\abbrev{BLAST}{Basic Local Alignment Search Tool}
\abbrev{CNV}{Copy Number Variation}
\abbrev{CSS}{Cascading Style Sheet}
\abbrev{DNA}{Deoxyribonucleic Acid}
\abbrev{DOM}{Document Object Model}
\abbrev{FASTA}{Fast All}
\abbrev{GFF}{General Feature Format}
\abbrev{HTML}{Hypertext Markup Language}
\abbrev{mRNA}{Messenger RNA}
\abbrev{RNA}{Ribonucleic Acid}
\abbrev{SNP}{Single Nucleotide Polymorphism}
\abbrev{SVG}{Scalable Vector Graphics}
\abbrev{XSS}{Cross Site Scripting}
}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% END OF FRONTMATTER SECTION
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{document}

% Typeset the title page
\maketitle

% Typeset the frontmatter.  
\frontmatter

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% FIRST CHAPTER OF THESIS BEGINS HERE
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\input{root/chapter_1_introduction.tex}
\input{root/chapter_2_related_work.tex}
\input{root/chapter_3_requirement_analysis.tex}
\input{root/chapter_4_visual_design.tex}
\input{root/chapter_5_synvisio.tex}
\input{root/chapter_6_evaluation.tex}
\input{root/chapter_7_discussion.tex}
\input{root/chapter_8_conclusion.tex}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% SUBSEQUENT CHAPTERS (or \input's)  GO HERE
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% The Bibliograpy should go here. BEFORE appendices!
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


% Typeset the Bibliography.  The bibliography style used is "plain".
% Optionally, you can specify the bibliography style to use:
% \uofsbibliography[stylename]{yourbibfile}
\uofsbibliography{reference.bib}

% If you are not using bibtex, comment the line above and uncomment
% the line below.  
%Follow the line below with a thebibliography environmentand bibitems.  
% Note: use of bibtex is usually the preferred method.

%\uofsbibliographynobibtex


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% APPENDICES
%
% Any chapters appearing after the \appendix command get numbered with
% capital letters starting with appendix 'A'.
% New chapters from here on will be called 'Appendix A', 'Appendix B'
% as opposed to 'Chapter 1', 'Chapter 2', etc.
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

% Activate thesis appendix mode.
\uofsappendix

% Put appendix chapters in the appendices environment so that they appear correcty
% in the table of contents.  You can use \input's here as well.
\begin{appendices}

\chapter{Exploring Conservation in Wheat}

To demonstrate SynVisio lets walk through the process of exploring conservation in Wheat. The data has been preloaded and can be accessed at \url{https://synvisio.usask.ca/#/Dashboard/ta_cs}.

\begin{figure}[h]
      \includegraphics[width=\textwidth]{images/appendix/step_1.PNG}
      \captionof{figure}{Select analysis mode}
\end{figure}

\begin{figure}[h]
      \includegraphics[width=\textwidth]{images/appendix/step_2.PNG}
       \captionof{figure}{Select default dashboard or an individual plot type}
\end{figure}

\begin{figure}[h]
      \includegraphics[width=\textwidth]{images/appendix/step_3.PNG}
       \captionof{figure}{Select track type for supplementary datasets}
\end{figure}

\begin{figure}[h]
      \includegraphics[width=\textwidth]{images/appendix/step_4.PNG}
      \captionof{figure}{Select source and target chromosomes which in this case belong to two sub genomes of wheat (A and B donors)}
\end{figure}

\begin{figure}[h]
      \includegraphics[width=\textwidth]{images/appendix/step_5.PNG}
      \captionof{figure}{Composite analysis dashboard showing conservation between two sub genomes of wheat (A and B donors) }
\end{figure}

\begin{figure}[h]
      \includegraphics[width=\textwidth]{images/appendix/step_6.PNG}
       \captionof{figure}{Toggle track visibility}
\end{figure}

\begin{figure}[h]
      \includegraphics[width=\textwidth]{images/appendix/step_7.PNG}
       \captionof{figure}{Filter conserved regions by gene count}
\end{figure}

\begin{figure}[h]
      \includegraphics[width=\textwidth]{images/appendix/step_8.PNG}
       \captionof{figure}{Select multi genome analysis and tree view}
\end{figure}

\begin{figure}[h]
      \includegraphics[width=\textwidth]{images/appendix/step_9.PNG}
       \captionof{figure}{Select chromosomes in each of the sub genomes of wheat.}
\end{figure}

\begin{figure}[h]
      \includegraphics[width=\textwidth]{images/appendix/step_10.PNG}
       \captionof{figure}{Tree view for multi genome analysis showing conservation between the three sub genomes of wheat (A, B, and D donors)}
\end{figure}

\begin{figure}[h]
      \includegraphics[width=\textwidth]{images/appendix/step_11.PNG}
       \captionof{figure}{Select multi genome analysis and hive view, then turn on normalized scales and chromosome labels}
\end{figure}

\begin{figure}[h]
      \includegraphics[width=\textwidth]{images/appendix/step_12.PNG}
       \captionof{figure}{Hive view showing conservation between the three sub genomes of wheat (A, B, and D donors) }
\end{figure}

\begin{figure}[h]
      \includegraphics[width=\textwidth]{images/appendix/step_13.PNG}
       \captionof{figure}{Highlight conserved regions emerging from each sub genome by clicking on the corresponding marker for that genome.}
\end{figure}


\end{appendices}

\end{document}
