\chapter{Related Work}

\section{Genomic Conservation and Synteny Detection}
In this section we discuss the biological background behind genomic conservation and how analysing it can provide crucial answers for researchers. We also explore the framework of synteny detection and some existing tools that are currently used in genomic research to perform the same.

\subsection{Biological Background}

Genomics is the field of biology that involves the study of genomes of various organism to understand their structure, function and evolution.\cite{world2002genomics}. A genome is defined as the complete set of DNA of an organism where DNA (Deoxyribonnucleic acid) is the chemical compound containing a series of instructions responsible for the development and functioning of that organism\cite{genomegov}.All living organisms transfer genomic information from one generation to the other through through chromosomes present in the nucleus of the cell. Humans for example, have 23 pairs of chromosomes where one from every pair is inherited from each parent and is responsible for their unique traits and characteristics. A chromosome structurally is a tightly packed length of DNA along with proteins that regulate its structure and activity. DNA is made of two long strings of nucleotide bases that are wrapped around each other in a double helix structure. There are four such bases: adenine (A), guanine(G), cytosine(C) and Thymine (T) with specific rules of pairing between them such that adenine always pairs with thymine and cytosine always pairs with guanine. These nitrogenous base pairs collectively make up the entire genome of an organism\cite{ussery2009computing}. The human genome for instance, is made up of around three billion base pairs encoding information for around 20,000-25000 genes\cite{international2004finishing}. 

Genes are long segments of DNA that encode information for a specific protein which are the basic building blocks of all organisms. Proteins are made up of long chain of amino acids where the structure and function of the proteins are determinant on the order of these amino acids. Fundamentally proteins are manufactured using the information encoded in a gene through the process of transcription and translation and collectively this process is called gene expression. During transcription the DNA present in a gene acts a template to form an mRNA which is a single stranded structure consisting of one of every complementary base pair in the DNA. This is followed by translation where mRNA is used as a template to assemble a chain of amino acids such that each group of three bases in the mRNA called a codon creates one particular amino acid. Thus the order of bases in the DNA encodes for the order of amino acids in the protein and in turns its structure and function\cite{clancy2008translation}.

DNA is transferred from one generation to the other in all livings organisms through the process of self replication where the double helical structure of the DNA comes apart and each of the complementary strands acts as a template in the production of its counterpart \cite{pray2008semi} forming new pairs of DNA strands. Although cellular error checking mechanisms ensure that these new DNA strands are nearly identical to the original strand, occasionally mutations can occur. This can happen when a base at one position is replaced by one of the other bases or is entirely entirely lost alternatively insertions or duplications of longer sets of base pairs can also happen. Other kinds of chromosomal rearrangements that occur include inversions, where a large segment of chromosome is reversed in orientation or translocation where parts of chromosomes swap places between themselves\cite{hartwell2008genetics}. While most mutations that occur during duplication do not have an affect on a gene they can however in some cases alter the function of the gene. This can be detrimental in some cases leading to diseases such as cancer.While in some other cases such mutations can be beneficial by offering resistance to diseases or other environmental factors. 

\subsection{Comparative Genomics}
As these mutations accumulate over time they lead to the divergence of species. Understanding how these changes could have occurred is a major field of research in comparative genomics and has large scale implications such as improving the quality of human life\cite{collins2003vision}. Comparative genomics as the name suggests, involves comparing genome sequences of different species to identify regions of similarity and differences to gain information on the relatedness between the species genomically and functionally. The fundamental principle in comparative genomics remains simple in that sequences that encode for proteins and gene expression should be conserved in related species whereas sequences that are responsible for differences between species will themselves be divergent.

Comparative genomics can assist biologists in linking the phenotypic and genotypic properties of an organism to understand its different characteristics. For example researchers combined the gene expression data of several plants sequences which have higher gene duplication rates with evolutionary conservation data to improve on gene discovery \cite{hanada2008importance}.Also the comparative analysis of genes and their regulatory pathways in the context of phylogeny provides scientists with a better understanding of how evolution happens at a molecular level\cite{soltis2003role}. However the questions that are addressed by comparing genomes at different phylogentic distances can vary \cite{hardison2003comparative}. For example genomic comparison between species that are separated by very long phylogenetic differences such as yeast(\textit{Saccharomyces cerevisiae}), worms(\textit{Caenorhabditis
elegans}), and flies (\textit{Drosophila melanogaster}) reveals that their genomes encode for many of the same proteins while the order of the genes and sequences are not conserved\cite{rubin2000comparative}. Whereas, comparison between more closely related species like Humans(\textit{Homo sapiens}) and Chimpanzees(\textit{Pan troglodytes}) reveal that the overall divergence between the two genomes is only 4\% and results are more oriented towards identifying the differences\cite{varki2005comparing}. 

Comparative genomic studies are primarily focused around the study of homologous sequences which are gene sequences that have shared ancestry where the extent of homology is determined by sequence similarity and are commonly referred to as homologs. Such similarity between DNA sequences can occur either because of a speciation event (a species diverges into two separate species) leading to orthologs or due to a gene duplication event (a gene is duplicated within the same genome) leading to paralogs \cite{jensen2001orthologs}. Research into such similar genes especially in eukaryotic organisms can shed light on gene duplication events that lead to the creation of gene families which are large groups of gene sequences similar to each other and can also having a similar function or gene expression\cite{rubin2000comparative}. Usually when a gene duplication occurs the new gene either remains inactive as a pesudogene or exists as duplicate copy of the original gene performing the same function. Increased number of gene duplicates through natural selection can often lead to an increase in the protein synthesised by the gene. An example of this is the variation in gene copy number in the human salivary amylase gene (AMY1) responsible for starch hydrolysis in certain human populations\cite{perry2007diet}. A third scenario of gene duplication that can occur rarely is when the duplicated gene through mutations acquires a new function. An example of this is trocarin D gene of the Australian rough-scaled snake that acts as a toxin by coagulating the blood of its prey. Comparative genomic analysis of the trocarin D gene revealed that it is nearly identical to the coagulation factor(F) X gene present in the plasma of the snake responsible for blood coagulation to prevent bleeding when the snake is injured indicating that that the gene was recruited for a new function after a gene duplication event\cite{reza2007structure}.

\subsection{Synteny}
One of the ways in which homology can be inferred for understanding large scale duplication events is through studying collinearity of several genes, where both the gene content and order are conserved\cite{proost2011adhore}. Such long regions containing several genes that display collinearity in the order of Kilobases(Kb) to Megabases(Mb) are referred to as Synteny Blocks\cite{zeng2008orthocluster}. The word Synteny has greek origins with \textit{syn} meaning together and \textit{taenia} meaning ribbon and is used to indicate the presence of genetic loci on the same chromosome\cite{renwick1971mapping}. However synteny can also occur between different chromosomes and nowadays is more commonly  used to refer to ``gene loci in different organisms located on a chromosomal region of common evolutionary ancestry''\cite{passarge1999incorrect}.

With the availability of fully sequenced genomes of several model species, synteny analysis can reveal evolutionary adaptions and also improve the transfer of knowledge to non model organisms that haven't been fully mapped\cite{zhao2019network}. Synteny analysis particularly in angiosperms (flowering plants) can help in understanding the consequences of whole genome duplication in plant evolution\cite{adams2005polyploidy} as shown in the analysis of polyploidy in Thale cress(\textit{Arabidopsis thaliana})\cite{seoighe2003turning}. This state of polyploidy where an organism contains more than two sets of homologs is a widespread occurrence in plants  due to several whole genome duplication events at diverse temporal scales but is rare in mammals with evidence of the last whole genome duplication event occurring almost 500 million years ago\cite{adams2005polyploidy,panopoulou2005timing}. However gene collinearity is conserved to a greater order in mammals than plants thus making synteny analysis at smaller scales called microsynteny much more feasible\cite{zhao2019network}. In this way synteny analysis can be adapted to a wide range of scenarios and evolutionary scales based on the underlying question.

\subsection{Analysis Pipeline}

Synteny analysis consists of three major steps : Sequence Alignment, Synteny Detection and Data Visualization. Although SynVisio focuses on the last step we will take a brief look at the other preceding steps. Before analysing synteny between two more more organisms their genomes need to be sequenced and annotated at least partially into scaffolds. This is then followed by similarity detection between the two genomes through sequence alignment.

\subsubsection{Sequence Alignment}
Sequence alignment is extensively used in computational biology to assess the similarity between DNA,RNA, and protein sequences. Fundamentally sequence alignment works by arriving at an optimal alignment through a scoring mechanism, where gaps are introduced in one of both of the sequences but penalised accordingly. A gap at any position in the final alignment is an indication of an insertion or a deletion and is penalised because these events are far less likely to occur than mutations. The validity of sequence alignment results are dependant on the alphabet size of the sequences  as protein sequences can contain upto 20 different amino acids whereas DNA sequences only contain four different bases leading to better alignments in proteins.There are two major types of sequence alignments and they are used in different scenarios. The first type of sequence alignment where optimal match is found by aligning the two entire sequences end to end is called Global sequence alignment and is used to compare homologous sequences. The second type of alignment that looks at smaller sections or sub-sequences to find a match is called Local sequence alignment and is used in looking for patterns in a sequence when comparing to a larger set of sequences such as those in database. 

Every sequence alignment is centred around an optimization problem and early alignment techniques such as the Needleman and Wunsch Method \cite{needleman1970general} used a dynamic programming approach which is a computational strategy that breaks problems continuously into smaller sub problems and reuses the results of solved sub problems to arrive at a solution of the larger problem. A variation of this for local sequence alignment is the Smith-Waterman algorithm\cite{smith1981identification} that uses a matrix based  scoring scheme for comparing sub sequences.
However the time complexity of such methods increased exponential when searching for sequences in large databases leading to adoption of heuristic methods to align sequences such as the FASTA (Fast-All) algorithm \cite{lipman1985rapid}. Although this algorithm is no longer in use the name FASTA is still used for a popular file format in bioinformatics, for representing nucleotide and protein sequences as a series of single-letter characters.

BLAST (Basic Local Alignment Search Tool) is a popular local sequence alignment tool that acts as a direct successor to the FASTA algorithm being more time efficient and operating on the same file format\cite{pertsemlidis2001having}. It operates by identifying small query words that contains three nucleotides or amino acids for protein sequences in a particular order based on their occurrence along the sequence and closeness to other similar words. It then expands on these words on either direction based on searches from target databases rated by scoring matrix. Blast by default used BLOSUM62 (Block Substitution Matrix) as its scoring matrix which ensures that even more distantly related sequences are detected but other matrices such as PAM250 (Point Accepted Mutation) can also be specified.

\begin{figure}
  \centering
  \includegraphics[width=.75\linewidth]{images/ch_2_synteny_plots.jpg}
  \captionof{figure}{Different types of plots visualizing synteny generated by MCScanX : (A) dual synteny plot, (B) circle plot, (C) dot plot and (D) bar plot, From Wang et al.\cite{wang2012mcscanx}.}
  \label{fig:ch_2_synteny_plots}
\end{figure}

\subsubsection{Synteny Detection Tools}

The next step after detecting the similarity between two sequences is the actual synteny detection as alignment results are only pairwise between sequences and need to be grouped into larger blocks to look for patterns. Although Synteny detection tools differ in their operating file formats and computational efficiency they broadly work by combining positional information of genes along a genome sequence with pairwise BLAST results to construct chains of collinear gene pairs. Grouping neighboring gene pairs that match is one way of detecting syteny\cite{wang2012mcscanx} that is implemented in tools such as OrthoCluster\cite{zeng2008orthocluster},TEAM\cite{luc2003gene} and ADHoRe\cite{proost2011adhore}. These tools are however outdated and are not efficient in detecting syntenic blocks with conserved gene order especially in scenarios that might include chromosomal rearrangements and tandem duplications\cite{wang2012mcscanx}. A new class of synteny tools such as MCScanX\cite{wang2012mcscanx}, DAGChainer\cite{haas2004dagchainer} and CYNTENATOR\cite{rodelsperger2010cyntenator}, that utilize a dynamic programming approach to create chains of pair-wise collinear genes around anchor genes are much more efficient at detecting collinear syntenic blocks. Some such as MCScanX even offer downstream analysis tools with visualization results as shown in Figure \ref{fig:ch_2_synteny_plots}.     






\section{Genomic Visualizations} 

use cydneys paper for reference
\subsection{General visualizations} - JBrowse,GBrowse,Ensemble
\subsection{Synteny Vis systems} - VGSC,MizBee, Sybill

extensive description of jorges paper and its different features

\section{Interactivity in Visualizations}
\subsection{Zoom in to explore, search , visualization information seeking mantra}
\subsection{Multiple linked views}
\subsection{Snapshot visual state - revisitation support}
\subsection{Annotating Charts with Tracks}

