\chapter{Data and Task Abstraction}

SynVisio was developed over multiple iterations as the necessary data and requirements were refined through continuous consultation with our genome research collaborators. To understand the design choices for visual encoding and interaction that we adopted in developing SynVisio, we need to first elaborate on the underlying data and task abstractions. We start with the data abstraction, where we explain the different characteristics of the syntenic data and how it is computed using synteny detection tools and further processed by our system. We then explore the different analysis tasks that can be performed on syntenic data, and finally, we discuss supplementary requirements that help researchers in using and interacting with SynVisio.

\section{Data}
This section starts with the description of the structure of a genome and its constituent elements and follows through with an explanation of how syntenic data is generated and represented.

\subsection{Genome Structure and Scales}
The genome of every organism is unique and can be defined as the complete set of DNA consisting of a series of instructions needed to build and maintain that organism. Structurally genomes are broken down into smaller sections called chromosomes, where every chromosome is a long strand of DNA coiled up along with various proteins. Each chromosome is made up of several genes, which are the basic functional units of heredity and code for a specific protein. Genes can then be further broken down into nucleotides which are the smallest building blocks of DNA. For analyzing genomic conservation, researchers also look at a collection of collinear genes that are called blocks. Thus the genome structure can be ordered into the following five levels from top to bottom in terms of genomic size : \textit{Genome\textrightarrow{Chromosome}\textrightarrow{Block} \textrightarrow{Gene}\textrightarrow{Nucleotide}}. However, to analyze large scale genomic conservation we only look at the first four levels. The structural data of a genome describing its constituent entities is provided to us in the form of a \textit{GFF(General Feature Format) File}. It contains the start and end position of every gene on a linear scale in a chromosome, the gene identifier, and the reference name of the parent chromosome in a three tab-delimited format. A sample GFF file can be seen in figure \ref{fig:ch_3_gff_file}.  Information on several genomes belonging to different species can be present in the same file and is distinguished by the two-character key present in the reference name of the chromosomes. The data on this file is processed by SynVisio to get the genomic size (number of nucleotides) of the different genomes and their constituent chromosomes along with the size and position of all the genes within these chromosomes. This data collectively gives us a precise structural map of every genome and its different sub-elements and so can be used to visualize a genome over multiple scales/levels.

\begin{figure}[h]
  \centering
  \includegraphics[width=.55\linewidth]{images/ch_3_gff_file.PNG}
  \captionof{figure}{Sample GFF File Describing Structure of a Genome.}
  \label{fig:ch_3_gff_file}
\end{figure}


\begin{figure}[h]
  \centering
  \includegraphics[width=.675\linewidth]{images/ch_3_coll_file.PNG}
  \captionof{figure}{Sample Collinearity File with a single block highlighted.}
  \label{fig:ch_3_coll_file}
\end{figure}

\begin{figure}[h]
  \centering
  \includegraphics[width=.45\linewidth]{images/ch_3_track_file.PNG}
  \captionof{figure}{Sample Track File.}
  \label{fig:ch_3_track_file}
\end{figure}


\subsection{Conservation Data}
At the smallest level, conservation between two genes can be inferred by looking at sequence homology, which is the similarity between the nucleotide sequences. Larger genomic conservation events can be studied by looking at blocks of such homologous genes and grouping them together based on their chromosomal positions to identify collinearity. To arrive at this data, we first identify all the homologous genes between two genomes using a local alignment tool such as BLAST (Basic Local Alignment Search Tool)\cite{blasttool}. Different synteny detection tools then construct collinear blocks of these homologous genes by either clustering neighbour matching gene pairs (DAGChainer) \cite{haas2004dagchainer} or by constructing chains around an anchor gene (MCScanX)\cite{wang2012mcscanx}. These collinear blocks also referred to as syntenic blocks, are the primary source of data for our tool and are provided as a collinearity file. A sample collinearity file can be seen in figure \ref{fig:ch_3_coll_file}. Every block of collinear genes has a corresponding similarity score indicating the level of match; an expect value (\textit{e-value}) indicating the probability that the match may have been due to chance; the count of genes; the names of the source and target chromosomes the blocks of genes belong to; and the orientation of the block(forward or reverse). Finally, every block of data also consists of a list of the homologous gene pairs in that block and their statistical significance(\textit{e-value}). This data, combined with the information on the structure of the genome can be used to associate every region of a genome with its homologous regions in the other genome or within itself depending on the type of synteny under investigation.

\subsection{Auxiliary Track Data}
Since genomic data is represented as a collection of linear sequences, additional information can be provided in the form of tracks parallel to the original gene sequence structure. These tracks can contain information about genomic features such as gene density, copy number variations (CNVs), and single-nucleotide polymorphisms (SNPs). The data is provided to SynVisio in a BedGraph file format consisting of 4 tab-delimited values: chromosome identifier, chromosomal start position, chromosomal end position, and a data value. This information can be hierarchically grouped in the same way as genomic sequences into Block Level or Chromosomal Levels and can be used to annotate the corresponding sequence structure. 


\section{Tasks}

In this section, we look at the different questions researchers have when analyzing genomic conservation. We then collate all the design requirements into a series of analysis tasks and group them based on the conservation relationship they address. Finally, we also elaborate on some additional elements that improve the usability of the system and assist users in performing their analysis tasks.

\subsection{Requirement Gathering Phase}
To formulate our design requirements, we met with three groups of researchers working in distinct fields of biological research. The meetings were conducted over several iterations to refine our initial requirements. The sessions broadly revolved around understanding the basic tasks researchers perform when analyzing genomic conservation and looking at the shortcomings of the existing synteny analysis tools in the market. Although all three groups were primarily interested in analyzing synteny, their individual use cases varied, providing us with a diverse set of user scenarios.

Our first research group was involved in researching genomic conservation in the \textit{brassica} genus as it offers an ideal model to study polyploid evolution which is responsible for genetic variations that are advantageous from an evolutionary perspective\cite{madlung2013polyploidy,liu2014brassica}. In particular they were interested in understanding genomic conservation within an allotetraploid species \textit{Brassica napus}(AACC) an important oilseed crop and also in comparing it to its closely related diploid species \textit{Brassica rapa}(AA) and \textit{Brassica oleracea}(CC) that belong together in the classical triangle of U\cite{nagaharu1935genome}. The requirements from this group were focused around having access to a system that could let them visualize the conserved relationship between the different diploid species and also within the chromosomes of a single allotetraploid (self synteny). 

The second research team we met were interested in looking at genomic conservation between \textit{Lens culinaris} and \textit{Cicer arietinum} to improve various agronomic traits as these are both widely grown legume crops. The requirements here were largely focused on cross synteny rather than self synteny. A unique trait of this particular dataset is that while \textit{C. arietinum} has a genome size of ~740 Mbp, \textit{L. culinaris} has a genome size of ~4 Gbp. This vast difference in the sizes of the two genomes makes visualizing synteny at the whole genome level extremely difficult and researchers must hence rely on comparing individual chromosomes of \textit{L. culinaris} one at a time or rely on a visualization that can have variable scales between the source and the target genomes.


The final research group we met with worked in sequencing wheat and were interested in understanding the genomic conservation between the three subgenomes of the hexaploid bread wheat genome. Researchers from this team wanted a system that could visualize synteny between 3 different genomes instead of a single source and target while taking into consideration the extremely large size of the wheat genome. They were also interested in adopting a novel network-based visualization called Hive Chart, which maps nodes on a radially distributed linear axes, for exploring multi-way genomic conservation.

\subsection{Visual Analysis Tasks}
Based on the underlying data and the requirements gathered from our research collaborators, we can group all analysis tasks performed in exploring genomic conservation into three basic groups according to the genomic scale at which they operate.

\smallskip
\noindent
\textbf{Genome Level}
\smallskip
\begin{enumerate}
\item [$Q_1$.] What is the level of conservation that exists between two or more sets of genomes?
\item [$Q_2$.] How does the density of conservation change across the genomes, and are there any gaps?
\item [$Q_3$.] How does the ordering of chromosomes based on conservation change between a given set of genomes or within a single genome? (possibility of detecting whole-genome duplication or genome reversal )
\item [$Q_4$.] If unmarked scaffolds exist, which regions of the target genome do they share similarity with?
\item[$Q_5$] Which chromosomes are sparsely or entirely unaligned, and how does the level of conservation change when these are ignored ?
\end{enumerate}

\smallskip
\noindent
\textbf{Chromosome Level}
\smallskip
\begin{enumerate}
\item [$Q_6$.]What is the level of conservation between a specific subset of chromosomes?
\item [$Q_7$.] What is the level of conservation between a single chromosome and an entire target genome or several other chromosomes (detecting unaligned regions within a chromosome)?
\item [$Q_8$.] 
How large are the collinear blocks relative to neighbouring chromosomes?
\item [$Q_9$.]What is the orientation of collinear blocks between two given chromosomes? (regular or inverted)
\end{enumerate}

\smallskip
\noindent
\textbf{Block Level}
\smallskip
\begin{enumerate}
\item [$Q_{10}$.]What is the level of conservation between the set of genes in a collinear block?
\item [$Q_{11}$.] What are the different genes contained in a block?
\item [$Q_{12}$.] What is the size of a gene relative to the size of the collinear block?
\item [$Q_{13}$.] Are there large gaps between genes in collinear blocks?
\item [$Q_{14}$.] How can $Q_{10}$-$Q_{13}$ be investigated when a collinear block is reversed?
\end{enumerate}

There are also several other analysis tasks that researchers perform that go beyond the block level into the nucleotide level. However, due to the limitations of the underlying data, we ignore these tasks as they are beyond the scope of this project, and there are several other systems specifically designed to investigate collinearity at the nucleotide level such as JBrowse \cite{skinner2009jbrowse}.


\subsection{Supplementary Design Requirements}
Apart from facilitating a system that can let researchers answer $Q_1$-$Q_{14}$ we also explore other design requirements that were raised during our discussions with our research collaborators. 

\textbf{Data Refinement} The underlying data that answers questions $Q_1$-$Q_9$ can be noisy due to the wrong parametric values used during synteny detection and can be filtered further based on several data attributes of the conserved blocks such as the similarity score, gene count and expect value (\textit{e-value}). This requires an interactive feature in the visual design of the system that can assist researchers in extracting a subset of the original dataset based on a particular threshold on a specific data attribute. Another direction in which this requirement can be extended, is having the ability to search for a specific gene in a dataset to look at all collinear blocks containing the gene.

\textbf{Data Enhancement} Analysis Tasks $Q_1$-$Q_9$ can be improved by providing a secondary data series that can contain additional information about the original sequences such as gene density and SNPs. This can also make it easy for researchers to identify patterns relative to known genomic markers.

\textbf{Data Exploration} Researchers need to have a system that offers multiple visual representations of the same information instead of relying on a single visualization to explore the richness of the data, shown by the different data attributes such as size, location , proximity and orientation of the conserved regions.

\textbf{Revisitation Support} The ability to simultaneously refine, enhance and explore complex data visualizations during synteny analysis can often overload the spatial cognition of the user causing them to lose context of their location in their workspace. This needs to be avoided and users need to be given a way to revisit the various steps in their synteny analysis pipeline when exploring a complex dataset.

\textbf{Web Access} In an increasingly connected world where research happens between a diverse set of collaborators spread across the globe, there is a rising need for synteny analysis systems to be available as web applications. Researchers need the final system to be accessed through a website that can let them directly upload the results of synteny detection tools to visualize the required information.




