\chapter{Related Work}

\section{Genomic Conservation and Synteny Detection}
In this section, we discuss the biological background behind genomic conservation and how analyzing it can provide crucial answers for researchers. We also explore the framework of synteny detection and some existing tools that are currently used in genomic research to perform the same.

\subsection{Biological Background}

Genomics is the field of biology that involves the study of genomes of various organisms to understand their structure, function, and evolution.\cite{world2002genomics}. A genome is defined as the complete set of DNA of an organism where DNA (Deoxyribonucleic acid) is the chemical compound containing a series of instructions responsible for the development and functioning of that organism\cite{genomegov}. All living organisms transfer this genomic information from one generation to the other through chromosomes present in the nucleus of the cell. Humans, for example, have 23 pairs of chromosomes where one from every pair is inherited from each parent and is responsible for their unique traits and characteristics. A chromosome structurally is a tightly packed length of DNA along with proteins that regulate its structure and activity. This DNA is made of two long strings of nucleotide bases along with sugar and phosphate groups that are wrapped around each other in a double helix structure. There are four such bases: adenine (A), guanine(G), cytosine(C) and Thymine(T) with specific rules of pairing between them such that adenine always pairs with thymine and cytosine always pairs with guanine. These nitrogenous base pairs collectively make up the entire genome of an organism\cite{ussery2009computing}. The human genome, for instance, is made up of around three billion base pairs encoding information for around 20,000-25000 genes\cite{international2004finishing}. 


\begin{figure}
  \centering
  \includegraphics[width=.75\linewidth]{images/ch_2_protein_generation.PNG}
  \captionof{figure}{Conversion of DNA information into protein via the genetic code. Complementary bases in a DNA strand are split into a single RNA strand, which is read in pairs of three bases at a time (codon) to create a single amino acid in a polypeptide chain. Partially adapted from \cite{hartwell2008genetics}.}
  \label{fig:ch_2_protein_generation}
\end{figure}


Genes are long segments of DNA that encode information for a specific protein, which are the basic building blocks of all organisms. Proteins are made up of long chains of amino acids where the structure and function of the proteins are determinant on the order of these amino acids. Fundamentally proteins are manufactured using the information encoded in a gene through the process of transcription and translation, and collectively, this process is called gene expression. During transcription, the DNA present in a gene acts as a template to form an mRNA, which is a single-stranded structure consisting of one of every complementary base pair in the DNA. This is followed by translation where mRNA is used as a template to assemble a chain of amino acids such that each group of three bases in the mRNA called a codon creates one particular amino acid, as shown in Figure \ref{fig:ch_2_protein_generation}. Thus the order of bases in the DNA encodes for the order of amino acids in the protein and, in turn, its structure and function\cite{clancy2008translation}.

DNA is transferred from one generation to the other in all livings organisms through the process of self-replication where the double-helical structure of the DNA comes apart, and each of the complementary strands acts as a template in the production of its counterpart forming new pairs of DNA strands\cite{pray2008semi}. Although cellular error-checking mechanisms ensure that these new DNA strands are nearly identical to the original strand, mutations can occasionally occur. This can happen when a base at one position is replaced by one of the other bases or is entirely lost. Alternatively, insertions or duplications of more extended sets of base pairs can also happen. Other kinds of larger mutations such as chromosomal rearrangements can occur, including inversions, where a large segment of a chromosome is reversed in orientation or translocation where parts of chromosomes swap places between themselves\cite{hartwell2008genetics}. While most mutations that occur during duplication do not have an affect on a gene, they can however, in some cases, alter the function of the gene. This can be detrimental in some cases, leading to diseases such as cancer. While in some other cases such mutations can be beneficial by offering resistance to diseases or other environmental factors. 

\subsection{Comparative Genomics}
As these mutations accumulate over time, they lead to the divergence of species. Understanding how these changes could have occurred is a significant field of research in comparative genomics and has large scale implications such as improving the quality of human life\cite{collins2003vision}. Comparative genomics, as the name suggests involves comparing genome sequences of different species to identify regions of similarity and differences to gain information on the relatedness between the species genomically and functionally. The fundamental principle in comparative genomics remains simple in that sequences that encode for proteins and gene expression should be conserved in related species, whereas sequences that are responsible for differences between species will themselves be divergent\cite{hardison2003comparative}.

Comparative genomics can assist biologists in linking the phenotypic and genotypic properties of an organism to understand its different characteristics. For example, researchers combined the gene expression data of several plant sequences which have high gene duplication rates with evolutionary conservation data to improve on gene discovery \cite{hanada2008importance}. Also, the comparative analysis of genes and their regulatory pathways in the context of phylogeny provides scientists with a better understanding of how evolution happens at a molecular level\cite{soltis2003role}. However, the questions that are addressed by comparing genomes at different phylogenetic distances can vary \cite{hardison2003comparative}. For example genomic comparison between species that are separated by very long phylogenetic differences such as yeast(\textit{Saccharomyces cerevisiae}), worms(\textit{Caenorhabditis
elegans}), and flies (\textit{Drosophila melanogaster}) reveals that their genomes encode for many of the same proteins while the order of the genes and sequences are not conserved\cite{rubin2000comparative}. Whereas, comparison between more closely related species like Humans(\textit{Homo sapiens}) and Chimpanzees(\textit{Pan troglodytes}) reveal that the overall divergence between the two genomes is only 4\% and results are more oriented towards identifying the differences than the similarities \cite{varki2005comparing}. 

Comparative genomic studies are primarily focused around the study of homologous sequences, which are gene sequences that have shared ancestry where the extent of homology is determined by sequence similarity and are commonly referred to as homologs. Such similarity between DNA sequences can occur either because of a speciation event (a species diverges into two separate species) leading to orthologs or due to a gene duplication event (a gene is duplicated within the same genome) leading to paralogs \cite{jensen2001orthologs}. Research into such similar genes, especially in eukaryotic organisms can shed light on gene duplication events that lead to the creation of gene families\cite{rubin2000comparative}. Gene families are defnied large groups of gene sequences that are similar to each other while also having a similar function or gene expression. Usually, when a gene duplication occurs the new gene either remains inactive as a pseudogene or exists as a duplicate copy of the original gene performing the same function. Increased number of gene duplicates through natural selection can often lead to an increase in the protein synthesized by the gene. An example of this is the variation in gene copy number in the human salivary amylase gene (AMY1) responsible for starch hydrolysis in certain human populations\cite{perry2007diet}. A third scenario of gene duplication that can occur rarely is when the duplicated gene through mutations acquires a new function. An example of this is trocarin D gene of the Australian rough-scaled snake that acts as a toxin by coagulating the blood of its prey. Comparative genomic analysis of the trocarin D gene revealed that it is nearly identical to the coagulation factor(F) X gene present in the plasma of the snake responsible for blood coagulation to prevent bleeding when the snake is injured indicating that that the gene was recruited for a new function after a gene duplication event\cite{reza2007structure}.

\subsection{Synteny}
One of the ways in which homology can be inferred for understanding large scale duplication events is through studying collinearity of several genes, where both the gene content and order are conserved\cite{proost2011adhore}. Such long regions containing several genes that display collinearity in the order of Kilobases(Kb) to Megabases(Mb) are referred to as Synteny Blocks\cite{zeng2008orthocluster}. The word Synteny has greek origins with \textit{syn} meaning together and \textit{taenia} meaning ribbon and is used to indicate the presence of genetic loci on the same chromosome\cite{renwick1971mapping}. However, synteny can also occur between different chromosomes and nowadays is more commonly used to refer to ``gene loci in different organisms located on a chromosomal region of common evolutionary ancestry''\cite{passarge1999incorrect}.

With the availability of fully sequenced genomes of several model species, synteny analysis can reveal evolutionary adaptions and also improve the transfer of knowledge to non-model organisms that haven't been fully mapped\cite{zhao2019network}. Synteny analysis, particularly in angiosperms (flowering plants), can help in understanding the consequences of whole genome duplication in plant evolution\cite{adams2005polyploidy} as shown in the analysis of polyploidy in Thale cress(\textit{Arabidopsis thaliana})\cite{seoighe2003turning}. This state of polyploidy where an organism contains more than two sets of homologs is a widespread occurrence in plants  due to several whole-genome duplication events at diverse temporal scales but is rare in mammals with evidence of the last whole genome duplication event occurring almost 500 million years ago\cite{adams2005polyploidy,panopoulou2005timing}. However gene collinearity is conserved to a greater order in mammals than plants thus making synteny analysis at smaller scales called microsynteny much more feasible\cite{zhao2019network}. In this way, synteny analysis can be adapted to a wide range of scenarios and evolutionary scales based on the underlying question.

\subsection{Analysis Pipeline}

Synteny analysis consists of three major steps: Sequence Alignment, Synteny Detection, and Data Visualization. Although SynVisio focuses on the last step, we will take a brief look at the other preceding steps. Before analyzing synteny between two or more organisms their genomes need to be sequenced and assembled at least partially into scaffolds. This process is then followed by similarity detection between the two genomes through sequence alignment.



\subsubsection{Sequence Alignment}
Sequence alignment is extensively used in computational biology to assess the similarity between DNA, RNA, and protein sequences. Fundamentally sequence alignment works by arriving at an optimal alignment through a scoring mechanism, where gaps are introduced in one of both of the sequences but penalized accordingly, as shown in Figure \ref{fig:ch_2_sequence_alignment_multi}. A gap at any position in the final alignment is an indication of an insertion or a deletion and is penalized because these events are far less likely to occur than mutations. The validity of sequence alignment results is dependant on the alphabet size of the sequences  as protein sequences can contain up to 20 different amino acids, whereas DNA sequences only contain four different bases leading to better alignments in proteins. There are two major types of sequence alignments, and they are used in different scenarios. The first type of sequence alignment where an optimal match is found by aligning the two entire sequences end to end is called Global sequence alignment and is used to compare homologous sequences. The second type of alignment that looks at smaller sections or sub-sequences to find a match is called Local sequence alignment and is used in looking for patterns in a sequence when comparing it with a larger set of sequences such as those in a database. 

\begin{figure}
  \centering
  \includegraphics[width=.45\linewidth]{images/ch_2_sequence_alignment_multi.PNG}
  \captionof{figure}{Sequence alignment of sequences `TTCTAAGTG',`CTACTAAGG' and `CTAATGTG' with mismatches and gaps highlighted in red and orange respectively.}
  \label{fig:ch_2_sequence_alignment_multi}
\end{figure}


Every sequence alignment is centred around an optimization problem and early alignment techniques such as the Needleman and Wunsch Method \cite{needleman1970general} used a dynamic programming approach to arrive at the optimal alignment. Dynamic programming is a computational strategy that recursively breaks larger problems into smaller sub problems and reuses the results of previously solved sub problems to arrive at a solution of the larger problem. A variation of the Needleman and Wunsch Method for local sequence alignment is the Smith-Waterman algorithm\cite{smith1981identification} that uses a matrix based scoring scheme for comparing sub sequences.
However, the time complexity of such methods increases exponentially when searching for sequences in large databases leading to the adoption of heuristic methods to align sequences such as the FASTA (Fast-All) algorithm \cite{lipman1985rapid}. Although this algorithm is no longer in use, the name FASTA is still used for a popular file format in bioinformatics, for representing nucleotide and protein sequences as a series of single-letter characters.

BLAST (Basic Local Alignment Search Tool) is a popular local sequence alignment tool that acts as a direct successor to the FASTA algorithm being more time-efficient and operating on the same file format\cite{pertsemlidis2001having}. It operates by identifying small query words that contain three nucleotides or amino acids for protein sequences in a particular order based on their occurrence along the sequence and closeness to other similar words. It then expands on these words on either direction based on searches from target databases rated by scoring matrix. Blast by default uses BLOSUM62 (Block Substitution Matrix) as its scoring matrix which ensures that even more distantly related sequences are detected but other matrices such as PAM250 (Point Accepted Mutation) can also be specified.



\subsubsection{Synteny Detection Tools}

The next step after detecting the similarity between two sequences is the actual synteny detection, as alignment results are only pairwise between sequences and need to be grouped into larger blocks to look for patterns. Although Synteny detection tools differ in their operating file formats and computational efficiency, they broadly work by combining positional information of genes along a genome sequence with pairwise BLAST results to construct chains of collinear gene pairs. Grouping neighboring gene pairs that match is one way of detecting syteny\cite{wang2012mcscanx} that is implemented in tools such as OrthoCluster\cite{zeng2008orthocluster},TEAM\cite{luc2003gene} and ADHoRe\cite{proost2011adhore}. These tools are, however, outdated and are not efficient in detecting syntenic blocks with conserved gene order, especially in scenarios that might include chromosomal rearrangements and tandem duplications\cite{wang2012mcscanx}. A new class of synteny tools such as MCScanX\cite{wang2012mcscanx}, DAGChainer\cite{haas2004dagchainer}, and CYNTENATOR\cite{rodelsperger2010cyntenator}, that utilize a dynamic programming approach to create chains of pairwise collinear genes around anchor genes are much more efficient at detecting collinear syntenic blocks. Some such as MCScanX even offer downstream analysis tools with visualization results, as shown in Figure \ref{fig:ch_2_synteny_plots}.     

\begin{figure}
  \centering
  \includegraphics[width=.75\linewidth]{images/ch_2_synteny_plots.jpg}
  \captionof{figure}{Different types of plots visualizing synteny generated by MCScanX : (A) dual synteny plot, (B) circle plot, (C) dot plot and (D) bar plot, From Wang et al.\cite{wang2012mcscanx}.}
  \label{fig:ch_2_synteny_plots}
\end{figure}

\section{Genomic Visualizations} 
With the advent of rapid genome sequencing systems, genomic data is being generated at an exponential pace that is not being met in terms of innovations in data analysis systems that can help researchers in understanding these ever increasing data streams. Visualization systems however can play a critical role in bridging this gap in data exploration as humans are intuitively good at finding patterns. As research in this field becomes increasingly data-driven, visualization systems can aid researchers, particularly in generating a hypothesis and iteratively refining it by encoding genomic information through visual cues in the form of shapes and colours\cite{nusrat2019tasks}. Genomic visualization systems can be used in several scenarios, such as analyzing sequence data at different resolutions and browsing annotations and reference tracks or in comparing sequences from different organisms\cite{nielsen2010visualizing}. However a major challenge in this field remains in determining the right graphical representation based on the genomic context and data under exploration. In this section, we first explore the different kinds of visualizations systems and techniques that are used in representing genomic data at the sequence and genome level. We then look particularly at systems that are used in exploring sequence similarity at different resolutions and finally give a brief overview of the current synteny visualization systems that are available and their merits and demerits.

\subsection{Sequence and Genome Browsers}
Visualizing genomes at the sequence level is primarily done by representing sequences
as a series of letters organized on a linear scale from left to right in the reading order. Additionally, sequences that are longer are stacked vertically. This ordering of bases or amino acids is meant to aid researchers in identifying discrepancies by quickly scanning down the sequence along its length, especially in scenarios involving comparing read alignments. Visual cues such as emphasis through colors are further used to highlight erroneous bases in some sequence analysis tools\cite{consed,ewing1998base}. Other tools like IGV (Integrative Genomics Viewer)\cite{thorvaldsdottir2013integrative} and Hawkeye\cite{schatz2007hawkeye} opt for a simpler representation by only visualizing discrepancies. Sequence visualization systems are also used to interpret and refine the results of sequencing systems where visual cues are provided to highlight gaps, mismatches, and the order of repeats\cite{bonfield1995new,consed}. In some systems such as Consed\cite{consed}, information is visualized in pairs that are color-coded so that structural variations such as insertions, deletions and inversions are also considered, and additional information is provided in the form of annotations of amino acids translations to identify miss-assemblies. Finally, some tools like the ABySS-Explorer provide an assembly graph overview for high-level inspection of the assembly instead of focusing on the local sequence mismatches\cite{nielsen2009abyss}. 


\begin{figure}
  \centering
  \includegraphics[width=1\linewidth]{images/ch_2_jbrowse.PNG}
  \captionof{figure}{JBrowse used to compare gene densities between (a) Barrel Medic(\textit{Medicago truncatula}), (b) Soybean(\textit{Glycine max}) and (c) Chickpea(\textit{Cicer arietinum}) in relation to curated genes encoding for prominent phenotypes (d) generated through KnowPulse \cite{sanderson2019knowpulse}. }
  \label{fig:ch_2_jbrowse}
\end{figure}


Once the short sequence reads are assembled into a single large genome, a different set of visualization tools are required that are focused around identifying particular regions of interest. A genome sequence in its entirety can acts a reference against which several features such as gene densities, SNPs, and repeats can be mapped. This form of analysis is increasingly being used in a wide range of browsers that were developed to disseminate information and provide a platform for the exploration of several genomes that are sequenced for model organisms such as GBrowse\cite{stein2002generic}, Ensembl Genome Browser\cite{stalker2004ensembl} and UCSC Genome Browser\cite{ucscgenome}. These browsers work by displaying a requested portion of the genome with several annotation tracks stacked vertically against each other along a reference axis. The annotation tracks can contain different kinds of information such as positions of single nucleotide polymorphisms, regions with regulatory elements, location of important genes, as shown in (d) of Figure \ref{fig:ch_2_jbrowse} and can be toggled on or off depending on their requirement in the analysis. The information is visualized at several resolutions from hundreds of base pairs all the way upto tens of thousands with the ability to move along the genome horizontally and zoom in and out of a particular region. Some genome browsers even offer the ability to search for a particular gene by looking up its position in the underlying database\cite{nielsen2010visualizing}. 


With an increase in the volume of data being generated, genome browsers are being adopted towards a decentralized model where processing power is distributed between the server and the client's browser. Information is retrieved only when needed for the resolution and the region that the user is interested in and visualizations are generated on the fly in the client-side. This reduces the load on the server substantially and modifies it to serve purely as a database to lookup information when necessary. This form of rendering in the client can offer a smoothly animated experience while navigating a genome through panning or zooming. It also ensures that there are no disruptions to a user's sense of location by averting incongruous transitions that occur when large data sets are being loaded and traversed. An example of such as browser is JBrowse that offers superior visualizations as shown in Figure \ref{fig:ch_2_jbrowse} along with significantly less server overhead compared to other genome browsers\cite{skinner2009jbrowse}.
Certain genome browsers take this decentralized model a step further by offering connectivity to data stored locally on a user's computer\cite{ucscgenome,saito2009utgb}. This can provide a personalized experience when dealing with sensitive data in situations where storing data on a remote server may not be viable. 

Finally, additional visual representations such as summary views (of copy number variations) in the context of biological pathways are also provided in certain browsers like UCSC Cancer genomics browser\cite{ucscgenome}. These genome browsers provide both sequence information and clinical information, allowing researches to associate clinical features with genomic data directly.


% \subsection{Compartive Genome Browsers}

% extensive description of jorges paper and its different features

% \section{Interactivity in Visualizations}
% \subsection{Zoom in to explore, search , visualization information seeking mantra}
% \subsection{Multiple linked views}
% \subsection{Snapshot visual state - revisitation support}
% \subsection{Annotating Charts with Tracks}

